certified-connectors/Tribal/apiDefinition.swagger.json

```diff
@@ -0,0 +1,2419 @@
+          "400": {
+            "description": "Bad Request"
+          },
+          "401": {
+            "description": "Unauthorized"
+          },
+          "403": {
+            "description": "Forbidden"
+          },
+          "500": {
+            "description": "Server Error"
+          }
+        }
+      }
+    },
+    "/api/events/filters/{eventType}/{entityType}": {
+      "get": {
+        "x-ms-visibility": "internal",
+        "description": "List of available filters",
+        "summary": "List of the available filters",
+        "operationId": "GetSchemaFilters",
+        "parameters": [
+          {
+            "name": "eventType",
+            "x-ms-summary": "The name of the event type",
+            "description": "The name of the event type",
+            "x-ms-url-encoding": "single",
+            "in": "path",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "entityType",
+            "x-ms-summary": "The name of the entity type",
+            "description": "The name of the entity type",
+            "x-ms-url-encoding": "single",
+            "in": "path",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/DynamicValue"
+              }
+            }
+          },
+          "401": {
+            "description": "Unauthorized"
+          },
+          "403": {
+            "description": "Forbidden"
+          },
+          "404": {
+            "description": "Not Found"
+          },
+          "500": {
+            "description": "Server Error"
+          }
+        }
+      }
+    },
+    "/api/events/schema/{eventType}/{entityType}": {
+      "get": {
+        "x-ms-visibility": "internal",
+        "description": "Gets the dynamic schema",
+        "summary": "Gets the dynamic schema",
+        "operationId": "GetDynamicSchema",
+        "parameters": [
+          {
+            "name": "eventType",
+            "x-ms-summary": "The name of the event type",
+            "description": "The name of the event type",
+            "x-ms-url-encoding": "single",
+            "in": "path",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "entityType",
+            "x-ms-summary": "The name of the entity type",
+            "description": "The name of the entity type",
+            "x-ms-url-encoding": "single",
+            "in": "path",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "type": "object"
+            }
+          },
+          "400": {
+            "description": "Bad Request"
```

Comment:
@mszy-tribal I can see that there are many response sections that are missing schema definitions, which may not be allowed. Can you please add schema definition for all the response properties?

Reply:
Is this a definite thing, I've noticed you have said "may not be allowed" implying that it could be. The reason is that the empty responses are from items which are private or are dynamically generated. 

I've been through the responses and I can see that the ones marked as "x-ms-visibility":"important" do have response properties where these will be used by our users. A couple of exceptions are the web hooks which are for the triggers to register which are not user facing.

I just want to make sure this is absolutely necessary otherwise it could be wasted effort.

Reply:
this has been done and the missing definitions have been added, the only exception is 204 because that is no content. The description has been altered to indicate that.